### PR TITLE
Change worker lockBlock to tryLockBlock

### PR DESCRIPTION
**Pull request by LuQQiu**
One of the potential fix/improve of #12444  and #11309 
Use tryLock() instead of lock() when locking worker blocks to prevent forever waiting for block locks.
When tryLock() failed (5 minutes by default), tryLock() is log a warning message and throw FailedToLockBlockException.

### CODE REVIEW

**Comment by LuQQiu** [comment]
Error: 9.869 [ERROR] alluxio.client.cli.fs.command.CatCommandIntegrationTest.alluxio.client.cli.fs.command.CatCommandIntegrationTest
Error: 9.872 [ERROR]   Run 1: CatCommandIntegrationTest>AbstractFileSystemShellTest.beforeClass:123 » Runtime
Error: 9.872 [ERROR]   Run 2: CatCommandIntegrationTest>AbstractFileSystemShellTest.afterClass:132 » NullPointer

File: core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java [hunk 1]
```diff
@@ -96,22 +106,65 @@ public BlockLockManager() {}
...
-                + " holds a lock on the block", sessionId, blockId));
+  public long tryLockBlock(long sessionId, long blockId, BlockLockType blockLockType)
+      throws FailedToLockBlockException {
+    Lock lock = getBlockLockWithType(sessionId, blockId, blockLockType);
+    int retryTime = 1;
+    boolean locked = false;
+    try {
+      while (retryTime <  mTryLockRetries) {
+        if (!lock.tryLock(TRY_LOCK_MIN, TimeUnit.MINUTES)) {
+          LOG.debug("Failed in attempt {}/{} to {} lock block of id {} and session {}",
```
Comment (maobaolong) [inline core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java:119]: When my alluxio worker getting low performance, or stuck, I will set this log level to `DEBUG`, and through this log, I still don't know which thread which session hold this block's lock and don't released, is there a way to show the lock holder information?

File: core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java [hunk 2]
```diff
@@ -96,22 +106,65 @@ public BlockLockManager() {}
...
+          locked = true;
+          break;
+        }
+        retryTime++;
       }
-      lock = blockLock.writeLock();
+    } catch (InterruptedException e) {
+      LOG.warn("Interrupted when trying to {} lock block of id {} and session {}",
+          blockLockType, blockId, sessionId);
+      lock.unlock();
```
Comment (maobaolong) [inline core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java:130]: Suppose that if there are other exception throw, the lock will not unlock?

**Comment by apc999** [comment]
replaced by https://github.com/Alluxio/alluxio/pull/12643/files

**Pull request closed without merging**